haystack_experimental/tools/component_tool.py

```diff
@@ -109,8 +130,9 @@ def __init__(self, component: Component, name: Optional[str] = None, description
             )
             raise ValueError(msg)
 
+        self.unresolved_parameters = parameters
```

Comment:
Just a thought, is this worth exposing? We could make this a private variable as well if it's only really useful for the `to_dict` method.